### PR TITLE
Advanced authoring tutorial

### LINKED ISSUE #230: Advanced Authoring Tutorial
Tutorial will include:
- creating Java @Action
- TBD

### PR DESCRIPTION
**Pull request by SamMarkowitz**
fixes #230

### CODE REVIEW

**Comment by SamMarkowitz** [comment]
@meshipeer  @Bonczidai @gituser173 @mihaitusa

Last chance for suggestions on what @Action to write in the tutorial. So far I have zero good ideas, so anything would be a help.

**Comment by levice14** [comment]
IMHO a simplified version of http call would do it

**Comment by SamMarkowitz** [comment]
Seems like HTTP is the consensus. I'll start running with it.

**Comment by SamMarkowitz** [comment]
@gituser173 please take a look at the first three lessons of the advanced authoring tutorial just to see if it makes sense as a starting point and to check that I'm doing things correctly.

**Comment by gituser173** [comment]
👍 looks good

File: advanced_authoring/02_lesson.rst [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+dependencies we'll be using in our project. We've started by declaring the score
+sdk, which contains some of the Java annotations we need, and JUnit, which we'll
+use for testing, as dependencies. We'll add more dependencies along the way as
+we need them.
+
+
+Class File
+----------
+
+Now that we have our project set up, let's start by create the class that will
```
Comment (gituser173) [inline advanced_authoring/02_lesson.rst]: "start by creating the class" I think

File: advanced_authoring/02_lesson.rst [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+@Action Annotation
+------------------
+
+Next, we'll add the ``@Action`` annotation within the class. This annotation
+allows CloudSlang to understand what information needs to be passed to the
+action method and what information it will receive from the action method.
+
+An @Action is made up of three parts. First, we provide a name for the action.
+Second, we declare an array of outputs for the action using the ``@Output``
+annotation. Third, we decalre an array of responses for the action using the
```
Comment (gituser173) [inline advanced_authoring/02_lesson.rst]: typo

File: advanced_authoring/02_lesson.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+We start by creating a map that will eventually contain the output values of the
+method (by convention, we call this map ``results``). Then we add a values for
+the two outputs and return the map.
+
+Tests
+-----
+
+Next, let's set up some tests. In the **src\\test\\java** folder let's create a new
+package named ``io.cloudslang.tutorial.httpclient.actions``. And within that
+package, create a new class named ``HttpClientActionTest``.
```
Comment (gituser173) [inline advanced_authoring/02_lesson.rst:183]: why name it HttpClientActionTest and HttpClientAction if the action doesn't do anything with HTTP?

File: advanced_authoring/03_lesson.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+put that value into the ``private`` variable ``forceFail`` which will be used by
+the Java method.
+
+Next we'll add the code to call the Java action:
+
+.. code-block:: yaml
+
+    java_action:
+        gav: 'io.cloudslang.tutorial:java-action:1.0-SNAPSHOT'
+        class_name: io.cloudslang.tutorial.actions.SaySomething
```
Comment (gituser173) [inline advanced_authoring/03_lesson.rst]: maybe use the class name of the action here

**Comment by gituser173** [comment]
@SamMarkowitz added a few comments, let me know if you need more help

File: advanced_authoring/02_lesson.rst (hunk 3 above)
Comment (SamMarkowitz) [reply advanced_authoring/02_lesson.rst:183]: The idea here is that we'll eventually build on it to do the HTTP stuff in later lessons. Do you think we should rename it to something more generic, or leave it as is?

**Comment by SamMarkowitz** [comment]
@gituser173 should I remove the lesson 4 and replace it with an "In progress"(or something similar) placeholder?

File: advanced_authoring/02_lesson.rst (hunk 3 above)
Comment (gituser173) [reply advanced_authoring/02_lesson.rst:183]: I understand, it's OK like it is now

**Comment by gituser173** [comment]
@SamMarkowitz sure 👍

**Comment by andreivasiletruta** [comment]
Closing as no plans to implement in the near future.

**Pull request closed without merging**